src/Bpi/ApiBundle/Controller/ChannelController.php

```diff
@@ -36,22 +38,84 @@ class ChannelController extends BPIController
      */
     public function listChannelsAction()
     {
-        $channelRepository = $this->getRepository('BpiApiBundle:Entity\Channel');
+        $query = new \Bpi\ApiBundle\Domain\Entity\ChannelQuery();
+        $query->amount(20);
+        if (false !== ($amount = $this->getRequest()->query->get('amount', false))) {
```

Comment:
Instead of $this->getRequest()->query->get() you can use getQueryParameter($parameterName) or getAllQueryParameters() from BPIController class.
https://github.com/inleadmedia/rest-api/blob/develop/src/Bpi/ApiBundle/Controller/ChannelController.php#L145